VAMobile/src/components/Switch.test.tsx

```diff
@@ -1,36 +1,20 @@
-import 'react-native'
-import { Switch as RNSwitch } from 'react-native'
 import React from 'react'
-// Note: test renderer must be required after react-native.
-import 'jest-styled-components'
-import { ReactTestInstance, act } from 'react-test-renderer'
-import Mock = jest.Mock
+import { fireEvent, screen } from '@testing-library/react-native'
 
-import { context, render, RenderAPI, waitFor } from 'testUtils'
+import { context, render } from 'testUtils'
 import Switch from './Switch'
 
 context('Switch', () => {
-  let component: RenderAPI
-  let testInstance: ReactTestInstance
-  let onPressSpy: Mock
+  let onPressSpy = jest.fn(() => {})
 
   beforeEach(() => {
-    onPressSpy = jest.fn(() => {})
-
-    component = render(<Switch onPress={onPressSpy} />)
-    testInstance = component.UNSAFE_root
-  })
-
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+    render(<Switch onPress={onPressSpy} a11yHint="Test switch" />)
   })
 
   describe('when the switch is clicked', () => {
-    it('should call the onPress function', async () => {
-      await waitFor(() => {
-        testInstance.findByType(RNSwitch).props.onValueChange()
-        expect(onPressSpy).toBeCalled()
-      })
+    it('calls the onPress function', async () => {
+      fireEvent.press(screen.getByA11yHint('Test switch'))
```

Comment:
Could alternatively be called with `getByRole`, but probably doesn't make much of a difference in this case
```tsx
screen.getByRole('switch')
```

Reply:
I decided to use `screen.getByRole('switch')` here, and add another test that checks for the a11yHint. That way we cover both bases